scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,553 @@
+                        f"toolchain {tc} is not available.\n"
+                        + "Please select from the list below:\n"
+                        + "\n".join(available_toolchains)
+                    )
+
+        installed_info = [v for (k, v) in self.fetch_sdk_info().items() if k == version]
+        if len(installed_info) == 0:
+            if args.install_dir:
+                base_dir = os.path.dirname(args.install_dir)
+                dir_name = os.path.basename(args.install_dir)
+            else:
+                base_dir = os.path.expanduser("~")
+                dir_name = None
+
+            sdk_dir = self.download_and_extract(
+                base_dir, dir_name, target_release, req_headers
+            )
+        else:
+            sdk_dir = installed_info[0]["path"]
+            self.inf(
+                f"Zephyr SDK version {version} is already installed at {sdk_dir}. Using it."
+            )
+
+        self.run_setup(args, sdk_dir)
+
+    def fetch_sdk_info(self):
+        sorted_sdks = []
+        try:
+            cmds = [
+                "-P",
+                os.path.join(os.path.dirname(__file__), "sdk", "listsdk.cmake"),
+            ]
+
+            output = zcmake.run_cmake(cmds, capture_output=True)
+            if output:
+                # remove '-- ' leader
+                sorted_sdks = [l[3:] for l in output]
+            else:
+                sorted_sdks = []
+
+        except Exception as e:
+            self.die(e)
+
+        sdk_info = {}
+        for sdk in reversed(sorted_sdks):
+            entry = {}
+
+            ver = None
+            if os.path.exists(os.path.join(sdk, "sdk_version")):
+                with open(os.path.join(sdk, "sdk_version")) as f:
+                    ver = f.readline().strip()
+            else:
+                continue
+
+            entry["path"] = sdk
+
+            if os.path.exists(os.path.join(sdk, "sysroots")):
+                entry["hosttools"] = "installed"
+
+            # Identify toolchain directory by the existence of <toolchain>/bin/<toolchain>-gcc
+            if "Windows" == platform.system():
+                gcc_postfix = "-gcc.exe"
+            else:
+                gcc_postfix = "-gcc"
+
+            toolchains = [
+                tc
+                for tc in os.listdir(sdk)
+                if os.path.exists(os.path.join(sdk, tc, "bin", tc + gcc_postfix))
+            ]
+
+            tcarray = []
+            if len(toolchains) > 0:
+                for tc in toolchains:
+                    tcarray.append(tc.strip())
+                entry["toolchains"] = tcarray
+
+            if ver:
+                sdk_info[ver] = entry
+
+        return sdk_info
+
+    def list_sdk(self):
+        sdk_info = self.fetch_sdk_info()
+
+        if len(sdk_info) == 0:
+            self.die("No Zephyr SDK installed.")
+
+        for k, v in sdk_info.items():
+            self.inf(k + ":")
+            self.inf("  path: " + v["path"])
```

Comment:
you may want to be more consistent in using formatted strings instead of concatenation.

Reply:
Fixed it.